File: src/main/java/de/ayont/lpc/moderation/Mute.java
package de.ayont.lpc.moderation;

import java.util.UUID;

public class Mute {
    private final UUID target;
    private final UUID sender;
    private final String reason;
    private final long expiry;

    public Mute(UUID target, UUID sender, String reason, long expiry) {
        this.target = target;
        this.sender = sender;
        this.reason = reason;
        this.expiry = expiry;
    }

    public UUID getTarget() {
        return target;
    }

    public UUID getSender() {
        return sender;
    }

    public String getReason() {
        return reason;
    }

    public long getExpiry() {
        return expiry;
    }

    public boolean isPermanent() {
        return expiry == -1;
    }

    public boolean hasExpired() {
        return !isPermanent() && System.currentTimeMillis() > expiry;
    }
}

File: src/main/kotlin/de/ayont/lpc/moderation/Mute.kt
package de.ayont.lpc.moderation

import java.util.UUID

data class Mute(
    val target: UUID,
    val sender: UUID,
    val reason: String,
    val expiry: Long
) {
    val isPermanent: Boolean
        get() = expiry == -1L

    fun hasExpired(): Boolean {
        return !isPermanent && System.currentTimeMillis() > expiry
    }
}
